### PR TITLE
Issue 4083: License Header is multi-line comment

### PR DESCRIPTION
**Pull request by TisonKun**
**Change log description**  

See also this [comment](https://github.com/pravega/pravega/pull/4021#discussion_r314118878) and description #4083 .

License header is not a kind of document comments indeed. Besides, it would confuse some IDEs like IDEA when reformatting the file. Change it into multi-line comment could solve the problem.

However, since we want to prevent style-only commit as well as the impact changing all license headers possibly non-functionally conflicts with a lot of ongoing efforts, we reach a consensus that just update `checkstyle/checkstyle.xml` for allowing multi-line comment style license header. It is hopefully we change license headers on the fly with other pull requests and we eventually adopt a multi-line style only codebase.

**Purpose of the change**  

Allow license header to be multi-line comment.

**What the code does**  

1. Replace `/\*\*\n \* Copyright \(c\)` with `/\*\*?\n * Copyright \(c\)` in `checkstyle/checkstyle.xml`

**How to verify it**  

trivial work.

ref #4083 

cc @andreipaduroiu @fpj

### CODE REVIEW

**Comment by TisonKun** [comment]
> A nice side effect is that if different contributors have different formatting settings, we won't be seeing a slew of formatting-only changes in our commit history - this helps reduce noise in PRs.

I can see your concerns. Also I locally use a workaround the same as yours :-)

If we decide not to rush in a huge commit then I would propose to change our checkstyle rules from `/\*\*\n \* Copyright \(c\)` to `/\*\(\*?)n \* Copyright \(c\)` and generally replace it on touched intentionally among further commits. What do you think about this approach?

**Comment by TisonKun** [comment]
Hi @andreipaduroiu I've updated the pull request. What do you think with the last proposal? Shall we keep context of this topic or just close it for a reason?

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java [hunk 1]
```diff
@@ -277,4 +277,4 @@ public void close() {
             tlsCertFileModificationMonitor.stopMonitoring();
         }
     }
-}
+}
```
Comment (fpj) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java]: This change is unrelated to this PR, please undo it.

File: checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
   <!-- header -->
   <module name="RegexpHeader">
-    <property name="header" value="/\*\*\n * Copyright \(c\)"/>
+    <property name="header" value="/\*\*?\n * Copyright \(c\)"/>
```
Comment (fpj) [inline checkstyle/checkstyle.xml]: The PR description no longer reflects this change. Could you please update, @TisonKun ?

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java (hunk 1 above)
Comment (TisonKun) [reply segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java]: Done.

File: checkstyle/checkstyle.xml (hunk 2 above)
Comment (TisonKun) [reply checkstyle/checkstyle.xml]: Updated

**Comment by TisonKun** [comment]
Thanks for your review @fpj . Addressed comment.

**Comment by TisonKun** [comment]
Any update?

**Comment by fpj** [comment]
hi @TisonKun , I'd rather not make this change. you are right that we could and perhaps even should have formatted the license header differently, but I'd rather not mix formats for the license header. we should keep this in mind for the future in the case we move the project under a different umbrella (_e.g._, ASF).

let me also say that I don't find it good practice to reformat existing files as you risk introducing unrelated formatting changes. pull requests, at least for this project, should not contain unrelated formatting changes. formatting changes should be in a separate pull request.

**Comment by TisonKun** [comment]
Thanks for your information.
